lib/trino-hive-formats/src/main/java/io/trino/hive/formats/line/openxjson/OpenXJsonDeserializer.java

```diff
@@ -311,14 +310,11 @@ void decodeValue(Object jsonValue, BlockBuilder builder)
 
             try {
                 long longValue = parseLong(jsonString.value());
-                if ((short) longValue == longValue) {
-                    SMALLINT.writeLong(builder, longValue);
-                    return;
-                }
+                SMALLINT.writeLong(builder, longValue);
```

Comment:
Same thing here about `SmallIntType#writeLong` throwing `TrinoException(GENERIC_INTERNAL_ERROR, ...)` instead of `BAD_DATA`